cpp/src/gandiva/engine.cc

```diff
@@ -147,7 +147,7 @@ Engine::Engine(const std::shared_ptr<Configuration>& conf,
 Status Engine::Init() {
   std::call_once(register_exported_funcs_flag, gandiva::RegisterExportedFuncs);
   // Add mappings for global functions that can be accessed from LLVM/IR module.
-  AddGlobalMappings();
+  ARROW_RETURN_NOT_OK(AddGlobalMappings());
```

Comment:
For external C interface functions, they may cause some error when adding mapping for them, so I change the `AddGlobalMappings` and `ExportedFuncsBase::AddMappings` functions to return `arrow::Status` to represent the result